core/azure-core/src/main/java/com/azure/core/polling/PollResponse.java

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Container of information related to polling and poll response. It will hold current polling status {@link OperationStatus} , T value,
+ * retry after duration and various configuration properties.
+ *
+ * <p><strong>Container of PollResponse and related information.</strong></p>
+ * @param <T> type of poll response value
+ */
+public final class PollResponse<T> {
+
+    private OperationStatus status;
+    private T value;
+    private Duration retryAfter;
+    private Map<Object, Object> properties;
```

Comment:
I would like all of these fields to be final so that the class can be considered immutable.

Reply:
I will make final except retryAfter  , properties because it is possible that are null. pollOperation does not provide that. We have a constructor which only takes PollResponse(status, T);

Reply:
resolved